crates/compiler/can/src/desugar.rs

```diff
@@ -2104,28 +2132,58 @@ enum Step<'a> {
     Skip,
 }
 
+#[allow(clippy::too_many_arguments)]
```

Comment:
This annotation was already on a number of desugar functions, so I followed that trend and added some more. I'm open to refactoring, but I'm not totally sure what that would look like. Other parts of canonicalization use an `Env` struct which includes `arena`, `problems`, and `module_path`. If we used that struct in desugaring we could cut down a few args. Would it make more sense though to make an `Env` specific to desugaring?

Reply:
Yeah, it doesn't have to be in this PR, but I think it would totally make sense to make a desugaring `Env`. In fact, I think the other parts of canonicalization would benefit from moving more of their threaded arguments into their `Env`.